playbooks/authorized_key.yml

```diff
@@ -1,5 +1,42 @@
+# Original idea found at http://stackoverflow.com/a/39083724
+- name: Store known hosts of 'all' the hosts in the inventory file
+  hosts: localhost
+  connection: local
+  vars:
+    ssh_known_hosts_command: "ssh-keyscan -T 10"
+    ssh_known_hosts_file: "{{ lookup('env','HOME') + '/.ssh/known_hosts' }}"
+    ssh_known_hosts: "{{ groups['all'] }}"
+  tasks:
+  - name: For each host, scan for its ssh public key
+    shell: "ssh-keyscan -t rsa {{ hostvars[item].ansible_ssh_host }} 2>/dev/null"
+    with_items: "{{ ssh_known_hosts }}"
+    register: ssh_known_host_results
+    retries: 10
+    delay: 3
+    until: ssh_known_host_results.rc == 0
+    tags:
+      - ssh
+
+  - name: Remove the public key in the '{{ ssh_known_hosts_file }}'
+    known_hosts:
+      name: "{{ hostvars[item.item].ansible_ssh_host }}"
+      state: "absent"
+      path: "{{ ssh_known_hosts_file }}"
+    with_items: "{{ ssh_known_host_results.results }}"
+    tags:
+      - ssh
+
+  - name: Add/update the public key in the '{{ ssh_known_hosts_file }}'
+    known_hosts:
+      name: "{{ hostvars[item.item].ansible_ssh_host }}"
+      key: "{{ item.stdout }}"
+      state: "present"
+      path: "{{ ssh_known_hosts_file }}"
+    with_items: "{{ ssh_known_host_results.results }}"
+    tags:
+      - ssh
+
 - hosts: all
-  serial: 1
```

Comment:
Why remove this? That is to prevent more than one to be broken if the keys are wrong

Reply:
Ah, right that is good to keep then. I remove it as I thought it was there to allow you to say yes :slightly_smiling_face: But i re added then :+1: